### PR TITLE
[CORDA-1472]: Crackdown on warnings.

### PR DESCRIPTION
**Pull request by sollecitom**
https://r3-cev.atlassian.net/browse/CORDA-1472

### CODE REVIEW

File: node/src/test/kotlin/net/corda/node/services/statemachine/FlowFrameworkTests.kt [hunk 1]
```diff
@@ -156,7 +156,7 @@ class FlowFrameworkTests {
         bobNode.registerFlowFactory(WaitForOtherSideEndBeforeSendAndReceive::class) { NoOpFlow() }
         val sessionEndReceived = Semaphore(0)
         receivedSessionMessagesObservable().filter {
-            it.message is ExistingSessionMessage && it.message.payload is EndSessionMessage
+            it.message is ExistingSessionMessage && it.message.payload === EndSessionMessage
```
Comment (mnesbit) [inline node/src/test/kotlin/net/corda/node/services/statemachine/FlowFrameworkTests.kt]: I don't think this is equivalent. Worried that deserialization may make these non-equal references

File: node/src/main/kotlin/net/corda/node/services/statemachine/transitions/StartedFlowTransition.kt [hunk 2]
```diff
@@ -317,7 +317,7 @@ class StartedFlowTransition(
             val sessionState = checkpoint.sessions[sessionId]!!
             when (sessionState) {
                 is SessionState.Initiated -> {
-                    if (sessionState.initiatedState is InitiatedSessionState.Ended) {
+                    if (sessionState.initiatedState === InitiatedSessionState.Ended) {
```
Comment (mnesbit) [inline node/src/main/kotlin/net/corda/node/services/statemachine/transitions/StartedFlowTransition.kt]: Again I am a bit hesitant that === is the same as 'is' when we have deserialization.

File: node/src/main/kotlin/net/corda/node/services/statemachine/SingleThreadedStateMachineManager.kt [hunk 3]
```diff
@@ -336,7 +336,7 @@ class SingleThreadedStateMachineManager(
             val flowId = sessionToFlow[recipientId]
             if (flowId == null) {
                 deduplicationHandler.afterDatabaseTransaction()
-                if (sessionMessage.payload is EndSessionMessage) {
+                if (sessionMessage.payload === EndSessionMessage) {
```
Comment (mnesbit) [inline node/src/main/kotlin/net/corda/node/services/statemachine/SingleThreadedStateMachineManager.kt]: Not sure this is equivalent

File: node/src/main/kotlin/net/corda/node/services/schema/NodeSchemaService.kt [hunk 4]
```diff
@@ -87,9 +87,9 @@ class NodeSchemaService(extraSchemas: Set<MappedSchema> = emptySet(), includeNot
 
     // Because schema is always one supported by the state, just delegate.
     override fun generateMappedObject(state: ContractState, schema: MappedSchema): PersistentState {
-        if ((schema is VaultSchemaV1) && (state is LinearState))
+        if ((schema === VaultSchemaV1) && (state is LinearState))
```
Comment (mnesbit) [inline node/src/main/kotlin/net/corda/node/services/schema/NodeSchemaService.kt]: Don't think this works with Schema inheritance. Needs thought.

File: node/src/main/kotlin/net/corda/node/services/keys/KMSUtils.kt [hunk 5]
```diff
@@ -25,14 +25,12 @@ import java.time.Duration
...
  * @param issuerSigner a content signer for the issuer.
- * @param revocationEnabled whether to check revocation status of certificates in the certificate path.
  * @return X.509 certificate and path to the trust root.
  */
 fun freshCertificate(identityService: IdentityService,
                      subjectPublicKey: PublicKey,
                      issuer: PartyAndCertificate,
-                     issuerSigner: ContentSigner,
-                     revocationEnabled: Boolean = false): PartyAndCertificate {
+                     issuerSigner: ContentSigner): PartyAndCertificate {
```
Comment (mnesbit) [inline node/src/main/kotlin/net/corda/node/services/keys/KMSUtils.kt]: I think we will ultimately need this flag, even if not used today

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/protonwrapper/engine/ConnectionStateMachine.kt [hunk 6]
```diff
@@ -62,7 +62,7 @@ internal class ConnectionStateMachine(serverMode: Boolean,
         transport = Engine.transport()
         transport.idleTimeout = IDLE_TIMEOUT
         transport.context = connection
-        transport.setEmitFlowEventOnSend(true)
+        transport.isEmitFlowEventOnSend = true
```
Comment (mnesbit) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/protonwrapper/engine/ConnectionStateMachine.kt]: Sorry that doesn't give the right semantics, hence why I chose to use set.

File: finance/src/main/kotlin/net/corda/finance/contracts/CommercialPaper.kt [hunk 7]
```diff
@@ -42,7 +42,7 @@ import java.util.*
  *    to do this in the Apache BVal project).
  */
 
-val CP_PROGRAM_ID = "net.corda.finance.contracts.CommercialPaper"
+const val CP_PROGRAM_ID = "net.corda.finance.contracts.CommercialPaper"
```
Comment (mnesbit) [inline finance/src/main/kotlin/net/corda/finance/contracts/CommercialPaper.kt]: Do we still need these strings?

File: testing/smoke-test-utils/src/main/kotlin/net/corda/smoketesting/NodeConfig.kt [hunk 8]
```diff
@@ -35,7 +36,7 @@ class NodeConfig(
                         .withValue("address", addressValueFor(rpcPort))
                         .withValue("adminAddress", addressValueFor(rpcAdminPort))
                         .root())
-                .withValue("rpcUsers", valueFor(users.map(User::toMap).toList()))
+                .withValue("rpcUsers", valueFor(users.map { it.toConfig().root().unwrapped() }.toList()))
```
[the remark this answers is not in the document]
Comment (sollecitom) [reply testing/smoke-test-utils/src/main/kotlin/net/corda/smoketesting/NodeConfig.kt]: It's deprecated

File: samples/trader-demo/src/main/kotlin/net/corda/traderdemo/TransactionGraphSearch.kt [hunk 9]
```diff
@@ -62,7 +62,7 @@ class TransactionGraphSearch(private val transactions: TransactionStorage,
                 results += tx
 
             val inputsLeadingToUnvisitedTx: Iterable<StateRef> = tx.inputs.filter { it.txhash !in alreadyVisited }
-            val unvisitedInputTxs: Map<SecureHash, SignedTransaction> = inputsLeadingToUnvisitedTx.map { it.txhash }.toHashSet().map { transactions.getTransaction(it) }.filterNotNull().associateBy { it.id }
+            val unvisitedInputTxs: Map<SecureHash, SignedTransaction> = inputsLeadingToUnvisitedTx.map { it.txhash }.toHashSet().mapNotNull { transactions.getTransaction(it) }.associateBy { it.id }
```
[the remark this answers is not in the document]
Comment (sollecitom) [reply samples/trader-demo/src/main/kotlin/net/corda/traderdemo/TransactionGraphSearch.kt]: `mapNotNull` does:

`Returns a list containing only the non-null results of applying the given [transform] function to each element in the original collection.`

File: samples/simm-valuation-demo/src/main/kotlin/net/corda/vega/plugin/customserializers/DoubleArraySerializer.kt [hunk 10]
```diff
@@ -2,11 +2,28 @@ package net.corda.vega.plugin.customserializers
 
 import net.corda.core.serialization.SerializationCustomSerializer
 import com.opengamma.strata.collect.array.DoubleArray
+import java.util.*
 
 @Suppress("UNUSED")
 class DoubleArraySerializer : SerializationCustomSerializer<DoubleArray, DoubleArraySerializer.Proxy> {
-    data class Proxy(val amount: kotlin.DoubleArray)
+    data class Proxy(val amount: kotlin.DoubleArray) {
```
[the remark this answers is not in the document]
Comment (sollecitom) [reply samples/simm-valuation-demo/src/main/kotlin/net/corda/vega/plugin/customserializers/DoubleArraySerializer.kt]: Well, you get the copy an all that, but if one of the fields is an array, data classes' equals is not correct. Not sure why they don't fix this.

File: samples/simm-valuation-demo/contracts-states/src/main/kotlin/net/corda/vega/contracts/PortfolioState.kt [hunk 11]
```diff
@@ -11,7 +11,7 @@ import java.time.LocalDate
 import java.time.ZoneOffset
 import java.time.temporal.ChronoUnit
 
-val PORTFOLIO_SWAP_PROGRAM_ID = "net.corda.vega.contracts.PortfolioSwap"
+const val PORTFOLIO_SWAP_PROGRAM_ID = "net.corda.vega.contracts.PortfolioSwap"
```
[the remark this answers is not in the document]
Comment (sollecitom) [reply samples/simm-valuation-demo/contracts-states/src/main/kotlin/net/corda/vega/contracts/PortfolioState.kt]: It's used in SimmValuationDemo.

File: node/src/main/kotlin/net/corda/node/services/schema/NodeSchemaService.kt (hunk 4 above)
Comment (sollecitom) [reply node/src/main/kotlin/net/corda/node/services/schema/NodeSchemaService.kt]: You can't inherit and object, can you?

File: node/src/main/kotlin/net/corda/node/services/keys/KMSUtils.kt (hunk 5 above)
Comment (sollecitom) [reply node/src/main/kotlin/net/corda/node/services/keys/KMSUtils.kt]: Reverted.

File: finance/src/main/kotlin/net/corda/finance/contracts/CommercialPaper.kt (hunk 7 above)
Comment (sollecitom) [reply finance/src/main/kotlin/net/corda/finance/contracts/CommercialPaper.kt]: Apparently we do :)

File: node/src/main/kotlin/net/corda/node/services/statemachine/SingleThreadedStateMachineManager.kt (hunk 3 above)
Comment (sollecitom) [reply node/src/main/kotlin/net/corda/node/services/statemachine/SingleThreadedStateMachineManager.kt]: As discussed.

File: samples/notary-demo/src/main/kotlin/net/corda/notarydemo/flows/DummyIssueAndMove.kt [hunk 12]
```diff
@@ -15,7 +15,7 @@ import net.corda.core.transactions.TransactionBuilder
 @StartableByRPC
 class DummyIssueAndMove(private val notary: Party, private val counterpartyNode: Party, private val discriminator: Int) : FlowLogic<SignedTransaction>() {
     companion object {
-        private val DO_NOTHING_PROGRAM_ID = "net.corda.notarydemo.flows.DummyIssueAndMove\$DoNothingContract"
+        private const val DO_NOTHING_PROGRAM_ID = "net.corda.notarydemo.flows.DummyIssueAndMove\$DoNothingContract"
```
[the remark this answers is not in the document]
Comment (sollecitom) [reply samples/notary-demo/src/main/kotlin/net/corda/notarydemo/flows/DummyIssueAndMove.kt]: It is :)

**Pull request merged**